pkg/labels/labels.go

```diff
@@ -131,21 +131,23 @@ func (ls Labels) Hash() uint64 {
 }
 
 // HashForLabels returns a hash value for the labels matching the provided names.
-func (ls Labels) HashForLabels(names ...string) uint64 {
-	b := make([]byte, 0, 1024)
-
-	for _, v := range ls {
-		for _, n := range names {
-			if v.Name == n {
-				b = append(b, v.Name...)
-				b = append(b, sep)
-				b = append(b, v.Value...)
-				b = append(b, sep)
-				break
-			}
+func (ls Labels) HashForLabels(b []byte, names ...string) (uint64, []byte) {
+	b = b[:0]
+	i, j, n1, n2 := 0, 0, len(names), len(ls)
+	for i < n1 {
```

Comment:
Could this be `for i:= range names` and remove the i++ at the end?  That'd make it more obvious that no trickery was going on inside the loop.

Reply:
And n1 seems to only be used here, so we could remove that.

Reply:
I tried with `i := range` first, but the traditional loop with `i++` cuts down more on time.